src/plots/cartesian/graph_interact.js

```diff
@@ -778,8 +778,9 @@ function createHoverText(hoverData, opts) {
     var i, traceHoverinfo;
     for (i = 0; i < hoverData.length; i++) {
         traceHoverinfo = hoverData[i].trace.hoverinfo;
-        if (traceHoverinfo.indexOf('all')===-1 &&
-                traceHoverinfo.indexOf(hovermode)===-1) {
+        var parts = traceHoverinfo.split('+');
```

Comment:
I like `isInFlagList` better (I like it a lot, in fact :+1: ) - `fullData` isn't supposed to *change* things from `data` unless they're invalid, it's just supposed to fill in the blanks.